simple/cq_data.c

```diff
@@ -481,7 +481,7 @@ int main(int argc, char **argv)
 
 	hints->ep_attr->type = FI_EP_MSG;
 	hints->caps = FI_MSG;
-	hints->mode = FI_LOCAL_MR | FI_PROV_MR_ATTR;
+	hints->mode = FI_LOCAL_MR | FI_PROV_MR_ATTR | FI_RX_CQ_DATA;
```

Comment:
Wait, isn't `FI_RX_CQ_DATA` a mode bit?  The issue here would be that the app should be testing for it on return from `fi_getinfo()` and using that to decide whether or not to post a recv to catch the incoming remote CQ data?